### ISSUE TITLE
Inconsistent comment wrapping

### ISSUE THREAD

**Issue by Baezon** [rust-lang/rustfmt#5329]
Using the rust playground and default rustfmt settings (version 1.4.38-nightly (2022-05-02 bed05e9)) this:
```
fn main() {
    println!("Hello, world! -------------------------------------------------"); //asd
    println!("Hello, world! -------------------------------------------------"); //asd
}
```

gets reformatted into this: 
```
fn main() {
    println!("Hello, world! -------------------------------------------------"); //asd
    println!("Hello, world! -------------------------------------------------"); 
    //asd
}
```

And there is no clear reason I can see why this would be reformatted in this way. For reference, the position after the semicolon is column 81, so it's within the default `max_width`. It *is* just past the default `comment_width`, even though `wrap_comments` is default off, maybe it's related?

**Comment by ytmimi** [comment]
Thanks for the report!

I can confirm that I'm able to reproduce the issue on [the playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=b04972b7f8b614059cd5c140e2e081d0).

Testing locally I wasn't able to reproduce the issue using the current master `rustfmt 1.4.38-nightly (b26c86b5 2022-04-30)`, which lead me to investigate non default configs used by the rustfmt `rustfmt.toml`, and I've tracked down that the issue only occurs when `version=One`. If you set `version=Two`, you get the correct formatting.

**Comment by ytmimi** [comment]
@calebcartwright should the `version=One` trailing comment placement be updated to be consistent with the `version=Two` formatting?

**Comment by calebcartwright** [comment]
> @calebcartwright should the `version=One` trailing comment placement be updated to be consistent with the `version=Two` formatting?

No, this change sounds like it was clearly version gated to avoid introducing a "breaking" formatting change and aligning with our formatting stability guarantee, and we shouldn't try to undo that.

I realize this is a bewildering and annoying formatting behavior, and I'd put it up there near the unnecessary spacing that gets inserted into nested tuple access by default. However, our hands are tied so we're stuck with this as the current default.

FWIW, I think we've finally got a path forward for evolving formatting in a way that'll satisfy all constraints, likely going into the next edition. In the interim though you'll either need to use `version=Two`, adjust your comment placement (e.g. above instead of after), or accept the current behavior.

Closing accordingly
